src/ln/router.rs

```diff
@@ -197,13 +200,15 @@ impl RoutingMessageHandler for Router {
 		secp_verify_sig!(self.secp_ctx, &msg_hash, &msg.bitcoin_signature_1, &msg.contents.bitcoin_key_1);
 		secp_verify_sig!(self.secp_ctx, &msg_hash, &msg.bitcoin_signature_2, &msg.contents.bitcoin_key_2);
 
-		//TODO: Call blockchain thing to ask if the short_channel_id is valid
-		//TODO: Only allow bitcoin chain_hash
-
 		if msg.contents.features.requires_unknown_bits() {
 			return Err(HandleError{err: "Channel announcement required unknown feature flags", action: None});
 		}
 
+		//TODO: Call blockchain thing to ask if the short_channel_id is valid
```

Comment:
Try to get this one done as described in the rfc at first but I think we need rust-blockchain-store API before. We're gonna need to access to UTXO set, built with rust-blockchain-store.